graylog2-web-interface/src/views/logic/search/GlobalOverride.ts

```diff
@@ -18,31 +18,37 @@ import * as Immutable from 'immutable';
 
 import type { QueryString, TimeRange } from '../queries/Query';
 
-export type MessageListOptions = {
-  [searchTypeId: string]: {
-    limit: number;
-    offset: number;
-  };
+export type PaginatedListOptions = {
+  limit: number,
+  offset: number,
+};
+
+export type ScrollListOptions = {
+  after: Array<any>,
+}
+
+export type SearchTypeOptions = {
+  [searchTypeId: string]: ScrollListOptions | PaginatedListOptions
```

Comment:
I do not think this is a good solution:
1. we are leaking plugin code here.
2. It does not take in the possibility that there might be other plugin authors which we do not know from.
3. Also the function itself does not care what is inside of `searchTypes`


```suggestion
  [searchTypeId: string]: any
```

Reply:
That is a good point, I tried to define the type definitions as generic as possible, but I am also fine with using `any` in this case.